### PR TITLE
Basic account onboarding

### PR DESCRIPTION
**Pull request by zachgoll**
Allows a user to select the "mode" their account's are in:

https://github.com/user-attachments/assets/c81495ff-5eb0-4f74-82c4-1d9246c6d7f9

Also includes a few bug fixes around syncing.

### CODE REVIEW

File: app/views/accounts/accountables/investment/_tabs.html.erb [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
+          <%= render "account/entries/loading" %>
+        <% end %>
+      <% when "transactions" %>
+        <%= turbo_frame_tag dom_id(account, :trades), src: account_trades_path(account) do %>
+          <%= render "account/entries/loading" %>
+        <% end %>
       <% end %>
+    <% else %>
+      <%= render "accounts/accountables/valuations", account: account %>
     <% end %>
```
Comment (zachgoll) [inline app/views/accounts/accountables/investment/_tabs.html.erb:32]: Will need to come back around and clean a lot of this markup up.  Lots of duplication across the accountable views

File: test/models/account/balance/syncer_test.rb [hunk 2]
```diff
@@ -47,6 +47,17 @@ class Account::Balance::SyncerTest < ActiveSupport::TestCase
...
+  test "syncs account with valuations and transactions when transaction starts" do
+    new_account = families(:empty).accounts.create!(name: "Test Account", balance: 1000, currency: "USD", accountable: Depository.new)
+    create_transaction(account: new_account, date: 2.days.ago.to_date, amount: 250)
+    create_valuation(account: new_account, date: Date.current, amount: 1000)
+
+    run_sync_for(new_account)
+
+    assert_equal 1000, new_account.balance
+    assert_equal [ 1250, 1000, 1000, 1000 ], new_account.balances.chronological.map(&:balance)
+  end
```
Comment (zachgoll) [inline test/models/account/balance/syncer_test.rb:59]: Validates bug when there is a transaction at an earlier date than the first valuation

**Comment by Muzosh** [comment]
I'm confused. What if I want to use both the way it was before? I.e. adding transactions for personal transaction logging, but also update value based on current's date account appreciation?

**Comment by Muzosh** [comment]
I would also argue that this splitting accounts into two modes is even more confusing. Let's say I created the account above, with "today's balance" 1 CZK. I then change my mind and convert this account from balance mode to transactions mode and add some transactions on the same day.

The value is not updated because the "today's balance" is still overwriting it.

I guess this is intented (although I could again argue that I added transaction **after** I updated today's balance on the same day) but now user has even lower chance of figuring out what is going on since he thinks his account is in "transactions only" mode.

**Comment by Muzosh** [comment]
Also if the account is in "balance only" mode, I can still successfully import transactions. However, after import it gets immediately overwritten by new balance entry, that has today's date and same value as previous entry.

**Comment by ttkjakub** [comment]
I was just thinking about the same thing today. I believe a solution here might be introducing a third value tracking mode called "Advanced" which would be a combination of Balance and Transaction tracking mode (= basically the old behavior).

What do you think @zachgoll?

**Comment by jakubkottnauer** [comment]
I was just thinking about the same thing today. I believe a solution here might be introducing a third value tracking mode called "Advanced" which would be a combination of Balance and Transaction tracking mode (= basically the old behavior).

What do you think @zachgoll?

**Comment by zachgoll** [comment]
> I was just thinking about the same thing today. I believe a solution here might be introducing a third value tracking mode called "Advanced" which would be a combination of Balance and Transaction tracking mode (= basically the old behavior).
> 
> 
> 
> What do you think @zachgoll?

This is in the works already!  It will just be a general "Activity View"

**Comment by zachgoll** [comment]
Working on an update that will look like this, which we sourced quite a bit of feedback on in Discord:

image

**Comment by Muzosh** [comment]
@zachgoll I just updated my instance and I'm super satisfied with this activity window. It solves all the issues I have written above :) Good job!

**Pull request merged**